rust/src/base.rs

```diff
@@ -539,6 +552,64 @@ impl FromStr for ConsignmentEndpoint {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, StrictEncode, StrictDecode)]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(crate = "serde_crate")
+)]
+#[non_exhaustive]
+pub enum InvoiceChain {
```

Comment:
I believe this should be not a `Chain` but rather a `Network`. The difference is the fact that you can have the same genesis (chain) but two different networks (Bitcoin, BCash etc). Also it is a network where the witness transaction is published, not a chain (to get into chain it has to be mined by miners).

And this brings the question why don't we use `bitcoin::Network` here.

Reply:
[`bitcoin::Network`](https://docs.rs/bitcoin/latest/bitcoin/network/constants/enum.Network.html) doesn't have the liquid network so I think we should keep our implementation, but I've renamed `InvoiceChain` to `Network` and the `chain` field to `network`